supervisor.go

```diff
@@ -154,41 +154,45 @@ func StartSupervisor(conf *Config) (Supervisor, error) {
 	// Spawn workers
 	var err error
 	for i := 0; i < conf.Provider.WorkerNum; i++ {
-		c, err := apns.NewConnection(conf.Apns.CertFile, conf.Apns.KeyFile, conf.Apns.SkipInsecure)
-
-		if err != nil {
-			LogWithFields(logrus.Fields{
-				"type": "supervisor",
-			}).Errorf("%s", err.Error())
-			break
-		} else {
-			LogWithFields(logrus.Fields{
-				"type":      "worker",
-				"worker_id": i,
-			}).Infoln("Succeeded to establish new connection.")
-			worker := Worker{
-				id:    i,
-				queue: make(chan Request, wqSize),
-				respq: make(chan SenderResponse, wqSize),
-				wgrp:  &sync.WaitGroup{},
-				sn:    conf.Apns.SenderNum,
-				ac: &apns.Client{
-					Host:   conf.Apns.Host,
-					Client: c,
-				},
-				fc: fcm.NewClient(conf.FCM.APIKey, nil, fcm.ClientTimeout),
+		var (
+			ac *apns.Client
+			fc *fcm.Client
+		)
+		if conf.Apns.enabled {
+			ac, err = apns.NewClient(conf.Apns.Host, conf.Apns.CertFile, conf.Apns.KeyFile, conf.Apns.SkipInsecure)
+			if err != nil {
+				LogWithFields(logrus.Fields{
+					"type": "supervisor",
+				}).Errorf("%s", err.Error())
+				break
 			}
-			LogWithFields(logrus.Fields{}).Infof("Response queue size: %d", cap(worker.respq))
-			LogWithFields(logrus.Fields{}).Infof("Worker Queue size: %d", cap(worker.queue))
-
-			s.workers = append(s.workers, &worker)
-			s.wgrp.Add(1)
-			go s.spawnWorker(worker, conf)
-			LogWithFields(logrus.Fields{
-				"type":      "worker",
-				"worker_id": i,
-			}).Debugf("Spawned worker-%d.", i)
 		}
+		if conf.FCM.enabled {
+			fc, err = fcm.NewClient(conf.FCM.APIKey, nil, fcm.ClientTimeout)
+			if err != nil {
+				LogWithFields(logrus.Fields{
+					"type": "supervisor",
+				}).Errorf("%s", err.Error())
+				break
+			}
+		}
+		worker := Worker{
+			id:    i,
+			queue: make(chan Request, wqSize),
+			respq: make(chan SenderResponse, wqSize),
+			wgrp:  &sync.WaitGroup{},
+			sn:    SenderNum,
```

Comment:
If `SenderNum` is defined as constant, I think that 'Worker' might not need `sn` field.
But, if the `sender_num` option at gunfish main are remained, `Worker` has `sn` field is ok.
Otherwise I think you delete theses (`sender_num` option and `sn` field) is all right.
https://github.com/kayac/Gunfish/blob/master/cmd/gunfish/gunfish.go#L38